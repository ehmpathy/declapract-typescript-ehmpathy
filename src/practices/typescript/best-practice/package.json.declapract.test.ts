import { contents } from './package.json.declapract';

describe('package.json', () => {
  it('should include build:artifact when project uses artifact practice', async () => {
    const declaredBestPracticeContents = await contents({
      projectPractices: ['artifact'],
    } as any);
    expect(declaredBestPracticeContents).toContain('build:artifact');
    expect(declaredBestPracticeContents).toMatchSnapshot();
  });
  it('should include not include build:artifact when project does not use artifact practice', async () => {
    const declaredBestPracticeContents = await contents({
      projectPractices: [],
    } as any);
    expect(declaredBestPracticeContents).not.toContain('build:artifact');
    expect(declaredBestPracticeContents).toMatchSnapshot();
  });
});
