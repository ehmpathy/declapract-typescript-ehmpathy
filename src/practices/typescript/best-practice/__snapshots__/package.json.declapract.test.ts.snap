// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`package.json should include build:artifact when project uses artifact practice 1`] = `
"{
  "devDependencies": {
    "typescript": "@declapract{check.minVersion('4.9.4')}",
    "@tsconfig/node-lts-strictest": "@declapract{check.minVersion('18.12.1')}"
  },
  "scripts": {
    "build:clean": "rm dist/ -rf",
    "build:compile": "tsc -p ./tsconfig.build.json",
    "build": "npm run build:clean && npm run build:compile && npm run build:artifact",
    "test:types": "tsc -p ./tsconfig.build.json --noEmit"
  }
}
"
`;

exports[`package.json should include not include build:artifact when project does not use artifact practice 1`] = `
"{
  "devDependencies": {
    "typescript": "@declapract{check.minVersion('4.9.4')}",
    "@tsconfig/node-lts-strictest": "@declapract{check.minVersion('18.12.1')}"
  },
  "scripts": {
    "build:clean": "rm dist/ -rf",
    "build:compile": "tsc -p ./tsconfig.build.json",
    "build": "npm run build:clean && npm run build:compile",
    "test:types": "tsc -p ./tsconfig.build.json --noEmit"
  }
}
"
`;
